src/Jobs/DeleteProducts.php

```diff
@@ -37,11 +37,6 @@ public function get_name(): string {
 	 * @throws ProductSyncerException If an error occurs. The exception will be logged by ActionScheduler.
 	 */
 	public function process_items( array $product_id_map ) {
-		$ready_ids = $this->product_repository->find_delete_product_ids( $product_id_map );
-
-		// Exclude any ID's which are not ready to delete.
-		$product_id_map = array_intersect( $product_id_map, $ready_ids );
-
```

Comment:
Ah, my brain wasn't working properly yesterday, I actually forgot to add back the deleted code so even though I added "status equals trash" logic in `find_delete_product_ids` method from `ProductRepository.php`, there is no one calling this method. So if I added it back like below It would try to find the product with status `trash` then exclude the ones which are not ready to delete. Note that `array_intersect( $product_id_map, $ready_ids );` can work as `array_intersect` is working through the array values.

```diff
--- a/src/Jobs/DeleteProducts.php
+++ b/src/Jobs/DeleteProducts.php
@@ -37,6 +37,12 @@ class DeleteProducts extends AbstractProductSyncerJob implements StartOnHookInte
   * @throws ProductSyncerException If an error occurs. The exception will be logged by ActionScheduler.
   */
  public function process_items( array $product_id_map ) {
+   $product_ids = array_values( $product_id_map );
+   $ready_ids   = $this->product_repository->find_delete_product_ids( $product_ids );
+
+   // Exclude any ID's which are not ready to delete.
+   $product_id_map = array_intersect( $product_id_map, $ready_ids );
+
    $product_entries = BatchProductIDRequestEntry::create_from_id_map( new ProductIDMap( $product_id_map ) );
    $this->product_syncer->delete_by_batch_requests( $product_entries );
  }
```

> but if we look at the sync hooks. It's hooked into both when a product is trashed and when it's deleted completely. In the latter case it won't be able to query the product data at all.

Correct me if I'm wrong, from the WooCommerce admin products page I can only delete a `trash` product completely. Which means based on the above code, we would call `delete_by_batch_request` when a product status is becoming `trash`, and we would not call `delete_by_batch_request` again when the product is deleted completely. Is it correct?

Reply:
A product can still be force deleted (without going to trash) either through WP-CLI, API or making use of a plugin.
In that case we have the ID mapping scheduled for removal, but we can't load any product details when the scheduled task actually starts running. So I imagine in that case the filter wouldn't work and the product would get excluded from the batch.

Reply:
Right, I think in this case it would be better to move this into different issue/PR so we don't block this PR. Do you think I should revert the commits 3f7a2cc and 7dfd462 then create an issue, or adding a new commit including the change in my [previous comment](https://github.com/woocommerce/google-listings-and-ads/pull/1728#discussion_r1024828110) and create an issue?

Reply:
I think fixing it temporarily by changing the status to trash (as long as we aren't reverting previous behaviour) is a good option. And then create a separate followup issue for filtering later, since that's covering behaviour that was already broken.

Reply:
The follow up issue is created in #1785.